src/script/lua_api/l_mapgen.cpp

```diff
@@ -1614,6 +1625,101 @@ int ModApiMapgen::l_generate_decorations(lua_State *L)
 }
 
 
+// generate_biomes(vm, p1, p2, [noise_filler_depth])
+int ModApiMapgen::l_generate_biomes(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	auto *mg_current = getMapgen(L);
+	auto *bgen = getBiomeGen(L);
+	if (!mg_current || !bgen)
+		return 0;
+
+	const NodeDefManager *ndef = mg_current->ndef;
+
+	MapgenBasic mg;
+
+	mg.c_stone              = ndef->getId("mapgen_stone");
+	mg.c_water_source       = ndef->getId("mapgen_water_source");
+	mg.c_river_water_source = ndef->getId("mapgen_river_water_source");
+
+	mg.vm   = checkObject<LuaVoxelManip>(L, 1)->vm;
+	mg.ndef = ndef;
+	mg.biomegen = mg_current->biomegen;
+	mg.biomemap = bgen->biomemap;
+	mg.water_level = mg_current->water_level;
```

Comment:
questionable whitespace alignment

```suggestion
	mg.vm          = checkObject<LuaVoxelManip>(L, 1)->vm;
	mg.ndef        = ndef;
	mg.biomegen    = mg_current->biomegen;
	mg.biomemap    = bgen->biomemap;
	mg.water_level = mg_current->water_level;
```

Reply:
I usually don't indent before  the "=" sign. Is this a Luanti code formatting policy?

Reply:
this wouldn't be a comment if not for the triple whitespace after `mg.vm`, and for how similar assignments are formatted in that same file. but yes, [this is part of code style guidelines](https://docs.luanti.org/for-engine-devs/code-style-guidelines/#spaces)

Reply:
Ooh, didn't know we have that.